arviz/plots/distcomparisonplot.py

```diff
@@ -134,6 +135,19 @@ def plot_dist_comparison(
         for data, var in zip(datasets, var_names)
     ]
 
+    total_plots = sum(
+        1 for _ in xarray_sel_iter(datasets[0], var_names=var_names[0], combined=True)
+    )
+    maxplots = len(dc_plotters[0]) * (len(groups) + 1)
+
+    if total_plots > rcParams["plot.max_subplots"]:
+        warnings.warn(
+            "rcParams['plot.max_subplots'] ({max_plots}) is smaller than the number "
+            "of variables to plot ({len_plotters}), generating only {max_plots} "
+            "plots".format(max_plots=maxplots, len_plotters=total_plots),
```

Comment:
This needs to be updated, not sure what is the best way to do so, but somehow. Here are the things that need to somehow be changed:

* `smaller than the number of variables to plot` does not work here. In most plots the number of subplots is the same as the number of variables, but here each variable has 3-4 subplots. Maybe using subplots instead of variable is enough?
* `rcParams['plot.max_subplots'] ({max_plots})` precisely because of this 3-4 factor, `max_plots` is not necessarily equal to the rcParam. I think it will be very confusing to users to set the rcParam to 7, then get a warning saying we'll only plot 6 subplots because you have set the rcParam to 6.